packages/lxd/src/file/pull.coffee.md

```diff
@@ -0,0 +1,139 @@
+# `nikita.lxc.file.pull`
+
+Pull files from containers.
+
+## Example
+
+```js
+const {$status} = await nikita.lxc.file.pull({
+  container: 'my_container',
+  source: './a_file',
+   target: `#{scratch}/a_file`
+})
+console.info(`File was pulled: ${$status}`)
+```
+
+## Todo
+
+* Ensure encrypting work as intended
+* Ensure gid and uid work as intended
+* Pull recursive directories
+* Detect name from lxd_source
+
+## Schema definitions
+
+    definitions =
+      config:
+        type: 'object'
+        properties:
+          'algo':
+            default: 'md5'
+            $ref: 'module://@nikitajs/core/lib/actions/fs/hash#/definitions/config/properties/algo'
+          'container':
+            $ref: 'module://@nikitajs/lxd/src/init#/definitions/config/properties/container'
+          'create_dirs':
+            type: 'boolean'
+            default: false
+            description: '''
+            Create any directories necessary.
+            '''
+          'gid':
+            type: ['integer', 'string']
+            description: '''
+            Set the file's gid on push, overwrite the `target` option.
+            '''
+          'lxd_source':
+            type: 'string'
+            description: '''
+            File source in the form of "[<remote>:]<container>/<path>",
+            overwrite the `source` option.
+            '''
```

Comment:
If we already have a container property, then why should we define the container name in `lxd_source`.

Reply:
I basically took the same approach as the `lxd.file.push` functionality that was already implemented. 

This property is the concatenation of both the remote (container name) and the path of the source. It is needed as the first argument of the `lxd file pull` command.

Reply:
This has been modified.

Reply:
Then in source, update the description with "Container path..." and mention that it is used inside `lxd_source`, update `lxc.file.push` as well to reflect the change, it is ok to include such changes along this merge request.

Reply:
`lxc.file.push` as been changed with specification on whether it is on the container side or on the local machine side.